frontend/src/components/ui/NewFromCurrent.jsx

```diff
@@ -0,0 +1,38 @@
+import { drawflowEditorAtom } from "@/atoms/drawflowAtom";
+import { pipelineAtom, pipelineFactory } from "@/atoms/pipelineAtom";
+import { HeaderMenuItem } from "@carbon/react";
+import { useAtom } from "jotai";
+import { trpc } from "@/utils/trpc";
+import { useWorkspace } from "@/hooks/useWorkspace";
+
+export default function NewFromCurrent() {
+  const [editor] = useAtom(drawflowEditorAtom);
+  const [pipeline] = useAtom(pipelineAtom);
+  const copyPipeline = trpc.copyPipeline.useMutation();
+  const { addPipeline } = useWorkspace();
+
+  const handleClick = async (editor, pipeline) => {
```

Comment:
This function already has access to the `editor` and `pipeline` so there shouldn't be a need to pass them as parameters.

Reply:
The reason for this is that the `editor` contains state that is not react watched, so having these as a closure can lead to stale data vs being passed as arguments:

https://javascript.plainenglish.io/stale-closures-in-react-afb0dda37f0b